pylibui/libui/entry.py

```diff
@@ -0,0 +1,138 @@
+"""
+ Python wrapper for libui.
+
+"""
+
+import ctypes
+from . import clibui
+
+
+class uiEntry(ctypes.Structure):
+    """Wrapper for the uiEntry C struct."""
+
+    pass
+
+
+def uiEntryPointer(obj):
+    """
+    Casts an object to uiEntry pointer type.
+
+    :param obj: a generic object
+    :return: uiEntry
+    """
+
+    return ctypes.cast(obj, ctypes.POINTER(uiEntry))
+
+
+# - char *uiEntryText(uiEntry *e);
+def uiEntryText(entry):
+    """
+    Returns the text of the entry.
+
+    :param entry: uiEntry
+    :return: string
+    """
+
+    clibui.uiEntryText.restype = ctypes.c_char_p
+    text = clibui.uiEntryText(entry)
+
+    return text.decode()
+
+
+# - void uiEntrySetText(uiEntry *e, const char *text);
+def uiEntrySetText(entry, text):
+    """
+    Sets the text of the entry.
+
+    :param entry: uiEntry
+    :param text: string
+    :return: None
+    """
+
+    clibui.uiEntrySetText(entry, bytes(text, 'utf-8'))
+
+
+# - void uiEntryOnChanged(uiEntry *e, void (*f)(uiEntry *e, void *data), void *data);
+def uiEntryOnChanged(entry, callback, data):
+    """
+    Executes the callback function on entry change.
+
+    :param entry: uiEntry
+    :param callback: function
+    :param data: data
+    :return: reference to C callback function
+    """
+    c_type = ctypes.CFUNCTYPE(
+        ctypes.c_int, ctypes.POINTER(uiEntry), ctypes.c_void_p)
+    c_callback = c_type(callback)
+
+    clibui.uiEntryOnChanged(entry, c_callback, data)
+
+    return c_callback
+
+
+# - int uiEntryReadOnly(uiEntry *e);
+def uiEntryReadOnly(entry):
+    """
+    Returns whether the entry is read only or not.
+
+    :param entry: uiEntry
+    :return: string
+    """
+
+    return bool(clibui.uiEntryReadOnly(entry))
+
+
+# - void uiEntrySetReadOnly(uiEntry *e, int readonly);
+def uiEntrySetReadOnly(entry, read_only):
+    """
+    Sets whether the entry is read only or not.
+
+    :param entry: uiEntry
+    :param read_only: bool
```

Comment:
`int` instead of `bool`. https://github.com/joaoventura/pylibui/pull/21/files#diff-28abfffa51d42558eda4a49aecd41b5fL60